src/regression.js

```diff
@@ -30,12 +30,26 @@
   /**
    * Determine the coefficient of determination (r^2) of a fit from the observations and predictions.
    *
+   * @param {Array<Array<number>>} data - Pairs of observed x-y values, including nulls
+   * @param {Array<Array<number>>} results - Pairs of observed predicted x-y values, including null predictions
+   *
    * @param {Array<Array<number>>} observations - Pairs of observed x-y values
    * @param {Array<Array<number>>} predictions - Pairs of observed predicted x-y values
    *
    * @return {number} - The r^2 value, or NaN if one cannot be calculated.
    */
-  function determinationCoefficient(observations, predictions) {
+  function determinationCoefficient(data, results) {
+    var observations = [];
+    var predictions = [];
+
+    // Remove predictive points for accurate fit
```

Comment:
For each null value in the observed data, there is also a prediction that has been filled in in the results. We have to filter out both.